Managers/PrimitiveMan.cpp

```diff
@@ -0,0 +1,230 @@
+#include "PrimitiveMan.h"
+#include "FrameMan.h"
+#include "SceneMan.h"
+#include "Entity.h"
+#include "MOSprite.h"
+
+#include "GUI.h"
+#include "AllegroBitmap.h"
+
+namespace RTE {
+
+	const std::string PrimitiveMan::c_ClassName = "PrimitiveMan";
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::GraphicalPrimitive::TranslateCoordinates(Vector targetPos, Vector scenePos, Vector & drawLeftPos, Vector & drawRightPos) {
+		drawLeftPos = scenePos;
+		drawRightPos = scenePos;
+
+		if (g_SceneMan.SceneWrapsX()) {
+			int sceneWidth = g_SceneMan.GetSceneWidth();
+			if (targetPos.m_X <= sceneWidth && targetPos.m_X > sceneWidth / 2) { targetPos.m_X = targetPos.m_X - sceneWidth; }
```

Comment:
Line 22 should be `... targetPos.m_X -= sceneWidth;` cause it's clear and easy to read.

Line 23 - Mixing assignment operators (i.e. =), especially compound assignment operators (e.g. -=) like this is super hard to read. It's like reading `float a = b += c;` <- wtf is going on there, right?
I'm not even sure what it's doing or if it works right cause it's bloody hard to follow. Also, for a guy who insists on totally pointless brackets in the condition part of ternaries, you skipped brackets where it actually matters, i.e. when order of operations isn't perfectly clear.

Anyway, how about:
```
drawLeftPos.m_X += (drawLeftPos.m_X > 0) ? -sceneHeight : (-sceneHeight + targetPos.m_X);
```
OR
```
drawLeftPos.m_X = (drawLeftPos.m_X > 0) ? (drawLeftPos.m_X - sceneWidth) : (drawLeftPos.m_X - sceneWidth + targetPos.m_X);
```

Same thing applies for the Y wrapping handling stuff below.